ignore docker push failures; barf on data races

tbg opened the pull request:
a docker push failing has been a source of red builds recently, so better to turn this off just for now. Can always make this a more elaborate retry loop later.

added logic to generally fail a test if anything ends up in `excerpt.log` (major player here: data races we never get to see except on `cockroach/master` because they don't fail the race test, whyever that happens).

tbg committed:
ignore failures when pushing to docker registry

tbg committed:
fail the test on data race warnings

tschottdorf commented:
See https://circleci.com/gh/cockroachdb/cockroach/1256?utm_campaign=build-fixed&utm_medium=email&utm_source=notification for a very recent example of a green build where `excerpt.txt` has a few data races for us.

tbg committed:
add a check for reply.Txn != nil

inspired by test failure in:
https://circleci.com/gh/tschottdorf/cockroach/510.

spencerkimball commented:
LGTM
